Src/PerformanceCollector/Shared/QuickPulseTelemetryModule.cs

```diff
@@ -0,0 +1,468 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.QuickPulse
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse;
+
+    using Timer = Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.Timer.Timer;
+
+    /// <summary>
+    /// Telemetry module for collecting QuickPulse data.
+    /// </summary>
+    public sealed class QuickPulseTelemetryModule : ITelemetryModule, IDisposable
+    {
+        private const int MaxSampleStorageSize = 10;
+
+        private readonly TimeSpan catastrophicFailureTimeout = TimeSpan.FromSeconds(5);
+
+        private readonly TimeSpan initialDelay = TimeSpan.Zero;
+
+        private readonly object lockObject = new object();
+
+        private readonly object collectedSamplesLock = new object();
+        
+        private readonly Uri serviceUriDefault = new Uri("https://qps.com/api");
+
+        private readonly LinkedList<QuickPulseDataSample> collectedSamples = new LinkedList<QuickPulseDataSample>();
+
+        private string instrumentationKey;
+
+        private IQuickPulseServiceClient serviceClient;
+
+        private Timer collectionTimer;
+
+        private Timer stateTimer;
+
+        private QuickPulseTimeProvider timeProvider;
+
+        private QuickPulseTimings timings;
+
+        private bool isInitialized;
+
+        private QuickPulseCollectionTimeSlotManager collectionTimeSlotManager = null;
+
+        private IQuickPulseDataAccumulatorManager dataAccumulatorManager = null;
+
+        private IQuickPulseTelemetryProcessor telemetryProcessor = null;
+
+        private QuickPulseCollectionStateManager stateManager = null;
+
+        private IPerformanceCollector performanceCollector = null;
```

Comment:
This will be addressed in future iterations. Only a few exceptions would be thrown though, since as per our existing perf collection logic we stop trying after a few failed attempts.